### PR TITLE
[DOC] Updates several tutorials

### PR DESCRIPTION
**Pull request by rrahn**
Changes:

* Reworked Getting Started Section
* Updated main page
* Added cover page for sequences
* Added cover page for dp alignment algorithms
* Cleaned Sequence tutorials
* Adde new tutorial for JST.

### CODE REVIEW

File: manual/source/Tutorial/GettingStarted/BackgroundAndMotivation.rst [hunk 1]
```diff
@@ -139,4 +141,102 @@ Such static polymorphism is different from **runtime polymorphism** which is sup
...
+Do you think you've got everything?
+Well, if not don't worry!
+Follow the :ref:`First Steps <tutorial-getting-started-first-steps-in-seqan>` tutorial which will cover the topics discussed above.
+This gives you the chance to apply the recently discussed paradigms to an actual (uhm, simplisitc) use case.
+But it will help you to better understand the way data structures and algorithms are implemented in SeqAn.
+
+We recommend you to also read the :ref:`Argument Parser Tutorial <tutorial-getting-started-parsing-command-line-arguments>`.
+This tutorial will teach you how to easily add command line arguments for your program and how to generate a help page for the options.
+Or you go back to the :ref:`main page <manual-main-tutorials>` and stroll through the other tutorials.
+You are know ready to dive deeper into SeqAn.
```
Comment (temehi) [inline manual/source/Tutorial/GettingStarted/BackgroundAndMotivation.rst]: know -> now

File: manual/source/Tutorial/GettingStarted/BackgroundAndMotivation.rst [hunk 2]
```diff
@@ -139,4 +141,102 @@ Such static polymorphism is different from **runtime polymorphism** which is sup
...
+Wow, this was quite some information to digest, wasn't it?
+We suggest you take a break!
+Get some fresh air!
+Grab something to drink or to eat!
+Let the information settle down.
+
+Do you think you've got everything?
+Well, if not don't worry!
+Follow the :ref:`First Steps <tutorial-getting-started-first-steps-in-seqan>` tutorial which will cover the topics discussed above.
+This gives you the chance to apply the recently discussed paradigms to an actual (uhm, simplisitc) use case.
```
Comment (temehi) [inline manual/source/Tutorial/GettingStarted/BackgroundAndMotivation.rst]: simplisitc - > simplistic

File: manual/source/Tutorial/GettingStarted/BackgroundAndMotivation.rst [hunk 3]
```diff
@@ -139,4 +141,102 @@ Such static polymorphism is different from **runtime polymorphism** which is sup
...
+Since this function only works for instances of :dox:`String String<`:dox:`AminoAcid AminoAcid>`, we could try to make it more general by making a template out of it.
+
+.. includefrags:: demos/tutorial/metafunctions/base.cpp
+    :fragment: func_exchange2
+
+Now the function works for all sequence types ``T`` that store ``AminoAcid`` objects, but it will fail for other value types as soon as the variable temp cannot store ``str[0]`` anymore.
+To overcome this problem, we must redefine ``temp`` in a way that it can store a value of the correct type.
+The question is: "Given a arbitrary type ``T``, what is the value type of ``T``?"
+
+The metafunction :dox:`ContainerConcept#Value` anwers this question: "The value type of ``T`` is given by ``Value<T>::Type``."
```
Comment (temehi) [inline manual/source/Tutorial/GettingStarted/BackgroundAndMotivation.rst]: anwers -> answers

File: manual/source/Tutorial/DataStructures/Sequence/StringsAndSegments.rst [hunk 4]
```diff
@@ -331,6 +311,234 @@ Assignment 4
...
+If no overflow strategy is specified, a default overflow strategy is selected depending on the type of the sequence.
+
+The following overflow strategies exist:
+
+:dox:`OverflowStrategyTags#Exact`
+  Expand the sequence exactly as far as needed. The capacity is only changed if the current capacity is not large enough.
+
+:dox:`OverflowStrategyTags#Generous`
+  Whenever the capacity is exceeded, the new capacity is chosen somewhat larger than currently needed.
+  This way, the number of capacity changes islimited in a way that resizing the sequence only takes amortized constant time.
```
Comment (temehi) [inline manual/source/Tutorial/DataStructures/Sequence/StringsAndSegments.rst]: islimited -> is limited

File: manual/source/Tutorial/DataStructures/Sequence/StringsAndSegments.rst [hunk 5]
```diff
@@ -331,6 +311,234 @@ Assignment 4
...
+
+.. includefrags:: demos/tutorial/sequences/base.cpp
+    :fragment: external_string_spec
+
+.. tip::
+
+   String Simplify Memory Management
+
+   One advantage of using Strings is that the user does not need to reserve memory manually with **new** and does not need **delete** to free memory.
+   Instead, those operations are automatically handeld by the :dox:`String` class.
```
Comment (temehi) [inline manual/source/Tutorial/DataStructures/Sequence/StringsAndSegments.rst]: handeld-> handled

File: manual/source/Tutorial/DataStructures/JournalStringTree.rst [hunk 6]
```diff
@@ -2,12 +2,233 @@
...
+
+#. The first data structure is a :dox:`Holder` which stores a sequence.
+#. The second data structure stores modifications that are made to this particular sequence using a **journal** (see `Journaling Filesystems <http://en.wikipedia.org/wiki/Journaling_file_system>`_ for more information).
+   This journal contains a list of deletions and insertions.
+   The inserted characters are stored in an additional **insertion buffer**.
+
+The advantage of this data structure lies in representing a String as a "patch" to another String.
+The journaled data structure can be modified without loosing the original context.
+We want to show you how to work with these data structures so you can build your own algorithms based on this.
+For this reason we keep the applicational background simple and implement an native online-search algorithm by which we examine different properties of the data structures.
```
Comment (temehi) [inline manual/source/Tutorial/DataStructures/JournalStringTree.rst]: applicational - application??
an -> a

File: manual/source/Tutorial/GettingStarted/BackgroundAndMotivation.rst (hunk 1 above)
Comment (rrahn) [inline manual/source/Tutorial/GettingStarted/BackgroundAndMotivation.rst]: done.

File: manual/source/Tutorial/GettingStarted/BackgroundAndMotivation.rst (hunk 2 above)
Comment (rrahn) [inline manual/source/Tutorial/GettingStarted/BackgroundAndMotivation.rst]: done

File: manual/source/Tutorial/GettingStarted/BackgroundAndMotivation.rst (hunk 3 above)
Comment (rrahn) [inline manual/source/Tutorial/GettingStarted/BackgroundAndMotivation.rst]: done

File: manual/source/Tutorial/DataStructures/Sequence/StringsAndSegments.rst (hunk 4 above)
Comment (rrahn) [inline manual/source/Tutorial/DataStructures/Sequence/StringsAndSegments.rst]: done.

File: manual/source/Tutorial/DataStructures/Sequence/StringsAndSegments.rst (hunk 5 above)
Comment (rrahn) [inline manual/source/Tutorial/DataStructures/Sequence/StringsAndSegments.rst]: done.

File: manual/source/Tutorial/DataStructures/JournalStringTree.rst (hunk 6 above)
Comment (rrahn) [inline manual/source/Tutorial/DataStructures/JournalStringTree.rst]: done.

**Comment by rrahn** [comment]
thanks :+1:

**Comment by xenigmax** [comment]
@temehi Can you merge it if you think it's fine?

**Comment by rrahn** [comment]
Please hold

**Comment by rrahn** [comment]
I will add some more changes to other tutorials.

**Comment by rrahn** [comment]
I'll notify you when it's done.

**Comment by temehi** [comment]
@rrahn this is not directly related to this PR. Since you have the files opened :pray: could you correct these typos as well.

file: BackgroundAndMotivation.rst
- Reuseability -> Reusability
- Reuseable -> Reusable
- library obtains genericity  ->  library stays generic (only a suggestion) 

file: StringsAndSegments.rst
- explicilty -> explicitly

**Comment by rrahn** [comment]
Ok, good catch @temehi :clap: 
I hope I covered everything now.

**Pull request merged**